src/middlewares/isWorkspaceMember.middleware.js

```diff
@@ -0,0 +1,30 @@
+const Channel = require("../models/channel.models");
+const Workspace = require("../models/workspace.models");
+const {
+    response_403,
+    response_500,
+    response_404,
+} = require("../utils/responseCodes.utils");
+
+exports.ensureUserIsMember = async (req, res, next) => {
+    try {
+        const user = req.body.user;
+        const workspace = await Workspace.findById(req.params.id);
+
+        if (!workspace) {
+            return response_404(res, "Workspace doesn't exist");
+        }
+
+        const isMember = workspace.channels.some(channel => channel.members.includes(user._id));
```

Comment:
You can not access channel.member

Reply:
Also admin of the workspace is a user 

You can not access channel.member

If the user is not member, send a response with status code 403 and message "You are not authorized to perform this action"A user is consider a member of a workspace if he is member of at least a channel belonging to it. Admin is a member also.

bro in this case I have to check members only right?
or I have to check createdBy(channels) is request param

Reply:
Can You explain this part alone? please @automatefreely